src/os/bluestore/BlueStore.cc

```diff
@@ -14590,18 +14591,20 @@ void BlueStore::_do_write_small(
     o->extent_map.punch_hole(c, offset, length, &wctx->old_extents);
 
     // Zero detection -- small block
-    if (!bl.is_zero()) {
+    if (!bl.is_zero() || !cct->_conf->bluestore_zero_block_detection) {
+      msg = " write small zero block ";
       BlobRef b = c->new_blob();
       _pad_zeros(&bl, &b_off0, min_alloc_size);
       wctx->write(offset, b, alloc_len, b_off0, bl, b_off, length, false, true);
     } else { // if (bl.is_zero())
-      dout(20) << __func__ << " skip small zero block " << std::hex
-        << " (0x" << b_off0 << "~" << bl.length() << ")"
-        << " (0x" << b_off << "~" << length << ")"
-        << std::dec << dendl;
+      msg = " skip small zero block ";
       logger->inc(l_bluestore_write_small_skipped);
       logger->inc(l_bluestore_write_small_skipped_bytes, length);
     }
+    dout(20) << __func__ << msg << std::hex
+      << " (0x" << b_off0 << "~" << bl.length() << ")"
+      << " (0x" << b_off << "~" << length << ")"
+      << std::dec << dendl;
```

Comment:
I'm not sure I follow the need to log the write (as opposed to skipped write).  It wasn't logged before -- presumably because many other existing log messages make it clear that the write is happening anyway?

Reply:
I want to be very clear about when the feature is used vs. not, especially since we still want to have the feature enabled in teuthology tests. Perhaps only one message is enough though. The absence of a "skip zero block" message will indicate that it has been written.

Reply:
Yeah, other (existing) log messages should be enough to indicate that the write wasn't skipped.